src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -588,8 +589,18 @@ public FormulaType GetNodeType(TexlNode node)
 
         // Called by language server to get custom tokens.
         // If binding is available, returns context sensitive tokens.  $$$
+        // Keeping this temporarily for custom publish tokens notification
+        // Feature to auto fix casing of function name in the editor also depends on this function and custom publish tokens notification
         internal IReadOnlyDictionary<string, TokenResultType> GetTokens(GetTokensFlags flags) => GetTokensUtils.GetTokens(this._binding, flags);
 
+        /// <summary>
+        /// Returns an enumeration of token text spans in a expression rule with their start and end indices and token type.
+        /// </summary>
+        /// <param name="comparer">Optional comparer to sort tokens.</param>
+        /// <param name="allowTokenHiding">Optional flag to indicate whether to compute whether token can be hidden or not.</param>

```

Comment:
> compute whether token can be hidden or not [](http://example.com/codeflow?start=79&length=43)

what does this mean?

Reply:
Each token has a CanBeHidden property. CanBeHidden property is used in old formula bar to hide (when it is true) that token in the editor when editor is not in focus (user is doing something else). For example, Color.Blue becomes Blue when user leaves the editor and again becomes Color.Blue when user comes back to editor. This is mostly UI feature as logically the formula is still Color.Blue. CanBeHidden property is computed for tokens created from first name nodes.  This flag allowTokenHiding basically indicates whether we want to compute the value of CanBeHidden property or not. We don't plan to support hiding tokens like old formula bar in the new formula bar so we don't need to spend time computing the value of CanBeHidden property as we would not use this. However, since we plan to replace the similar tokenization logic that backs the old formula bar, i added this allowTokenHiding here so i could enable computing the value of CanBeHidden property for tokens extracted out of first name nodes. I will try to improve the phrasing